### ISSUE TITLE
No rollback when aborting adding a madokami url

### ISSUE THREAD

**Issue by CounterPillow** [Hamuko/cum#3]
If a madokami URL is passed to 'get' and cum asks for the login information, but cum is prematurely terminated using a keyboard interrupt, a seemingly incomplete series is added. It cannot be unfollowed through the command line as it doesn't show up, and thus needs to be manually deleted from the database. Otherwise, the following exception is thrown if a user tries to update:

```python-traceback
Traceback (most recent call last):
  File "./cum.py", line 232, in <module>
    cli()
  File "/usr/lib/python3.4/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)                                                        
  File "/usr/lib/python3.4/site-packages/click/core.py", line 644, in main                   
    rv = self.invoke(ctx)                                                                    
  File "/usr/lib/python3.4/site-packages/click/core.py", line 991, in invoke                 
    return _process_result(sub_ctx.command.invoke(sub_ctx))                                  
  File "/usr/lib/python3.4/site-packages/click/core.py", line 837, in invoke                 
    return ctx.invoke(self.callback, **ctx.params)                                           
  File "/usr/lib/python3.4/site-packages/click/core.py", line 464, in invoke                 
    return callback(*args, **kwargs)                                                         
  File "./cum.py", line 227, in update                                                       
    series = series_by_url(follow.url)                                                       
  File "/home/fratti/Projekte/cum/scrapers/__init__.py", line 19, in series_by_url           
    return MadokamiSeries(url)                                                               
  File "/home/fratti/Projekte/cum/scrapers/madokami.py", line 19, in __init__
    self.chapters = self.get_chapters()
  File "/home/fratti/Projekte/cum/scrapers/madokami.py", line 45, in get_chapters
    chapter = name_parts.group(1)
AttributeError: 'NoneType' object has no attribute 'group'
```

**Comment by Hamuko** [comment]
>If a madokami URL is passed to 'get' and cum asks for the login information, but cum is prematurely terminated using a keyboard interrupt, a seemingly incomplete series is added.

Seems like a false diagnosis to me. I tried to reproduce this problem by adding a few series that I know to work with cum and using the keyboard interrupt at the login prompt. The result was that the series were added but no files were downloaded. Which makes sense considering that first the URL is scraped, then it's followed and lastly it looks up new chapters for it to download. The login prompt doesn't happen until it reaches the download part as the directory listing doesn't require login information.

What I'm guessing is that the initial scrape of the series failed and that it was still added to the DB as an incomplete entry. `AttributeError: 'NoneType' object has no attribute 'group'` suggests a regex error and when it comes to Madokami, it means that someone has given the files in the series retarded filenames.

What series did you try this with?
